sjsonnet/src/sjsonnet/Evaluator.scala

```diff
@@ -841,42 +841,40 @@ class Evaluator(
   def visitObjComp(e: ObjBody.ObjComp, sup: Val.Obj)(implicit scope: ValScope): Val.Obj = {
     val binds = e.preLocals ++ e.postLocals
     val compScope: ValScope = scope // .clearSuper
-
-    lazy val newSelf: Val.Obj = {
-      val builder = new java.util.LinkedHashMap[String, Val.Obj.Member]
-      for (s <- visitComp(e.first :: e.rest, Array(compScope))) {
-        lazy val newScope: ValScope = s.extend(newBindings, newSelf, null)
-
-        lazy val newBindings = visitBindings(binds, (self, sup) => newScope)
-
-        visitExpr(e.key)(s) match {
-          case Val.Str(_, k) =>
-            val prev_length = builder.size()
-            builder.put(
-              k,
-              new Val.Obj.Member(e.plus, Visibility.Normal) {
-                def invoke(self: Val.Obj, sup: Val.Obj, fs: FileScope, ev: EvalScope): Val =
-                  visitExpr(e.value)(
-                    s.extend(newBindings, self, null)
-                  )
+    val builder = new java.util.LinkedHashMap[String, Val.Obj.Member]
+    for (s <- visitComp(e.first :: e.rest, Array(compScope))) {
+      visitExpr(e.key)(s) match {
+        case Val.Str(_, k) =>
+          val prev_length = builder.size()
+          builder.put(
+            k,
+            new Val.Obj.Member(e.plus, Visibility.Normal) {
+              def invoke(self: Val.Obj, sup: Val.Obj, fs: FileScope, ev: EvalScope): Val = {
+                lazy val newScope: ValScope = s.extend(newBindings, self, sup)
+                lazy val newBindings = visitBindings(binds, (self, sup) => newScope)
+                visitExpr(e.value)(
+                  s.extend(newBindings, self, null)
```

Comment:
On closer look, I see that this is still passing `null` for `newSuper` here and that probably indicates that there's still a latent bug. Let me see if I can devise a test case.

Reply:
Indeed:

- Test: https://github.com/databricks/sjsonnet/pull/358/commits/55bcb2684a39a386981cb1f5ded712a92dcc3f82
- Fix: https://github.com/databricks/sjsonnet/pull/358/commits/fd038d84823f521c6213435730dcd373a7add26a